api/lib/opentelemetry/trace.rb

```diff
@@ -21,7 +21,7 @@ module Trace
     # the class `Random`.
     #
     # @return [#bytes]
-    RANDOM = Random.respond_to?(:bytes) ? Random : Random::DEFAULT
+    RANDOM = Random.respond_to?(:bytes) ? Random.new : Random::DEFAULT
```

Comment:
Unfortunately this will break "things that fork" (e.g. pre-forking web servers and pre-forking job servers). The problem is that the **global** random number generator is re-initialized by the Ruby VM on fork but instances are not. This results in an unfortunate situation where new traces generated by sibling processes generate the same sequence of trace IDs.

Reply:
aaah I see. thanks for the explanation! will close this 👍